shelley/chain-and-ledger/shelley-spec-ledger-test/test/Test/Shelley/Spec/Ledger/Rules/TestChain.hs

```diff
@@ -190,47 +221,72 @@ checkWithdrawlBound SourceSignalTarget {source, signal, target} =
 
 -- | If we are not at an Epoch Boundary, then (Utxo + Deposits)
 -- increases by Withdrawals min Fees (for all transactions in a block)
-utxoDepositsIncreaseByFeesWithdrawals :: SourceSignalTarget (CHAIN C) -> Property
+utxoDepositsIncreaseByFeesWithdrawals ::
+  ( ChainProperty era,
+    HasField "txfee" (Core.TxBody era) Coin,
+    HasField "wdrls" (Core.TxBody era) (Wdrl era)
+  ) =>
+  SourceSignalTarget (CHAIN era) ->
+  Property
 utxoDepositsIncreaseByFeesWithdrawals SourceSignalTarget {source, signal, target} =
   circulation target <-> circulation source
     === withdrawals signal <-> txFees signal
   where
     circulation chainSt =
       let es = (nesEs . chainNes) chainSt
           (UTxOState {_utxo = u, _deposited = d}) = (_utxoState . esLState) es
-       in balance u <+> d
+       in Val.coin (balance u) <+> d
 
 -- | If we are not at an Epoch Boundary, then (Utxo + Deposits + Fees)
 -- increases by sum of withdrawals for all transactions in a block
-potsSumIncreaseWdrlsPerBlock :: SourceSignalTarget (CHAIN C) -> Property
+potsSumIncreaseWdrlsPerBlock ::
+  ( ChainProperty era,
+    HasField "wdrls" (Core.TxBody era) (Wdrl era)
+  ) =>
+  SourceSignalTarget (CHAIN era) ->
+  Property
 potsSumIncreaseWdrlsPerBlock SourceSignalTarget {source, signal, target} =
   potsSum target <-> potsSum source === withdrawals signal
   where
     potsSum chainSt =
       let (UTxOState {_utxo = u, _deposited = d, _fees = f}) =
             _utxoState . esLState . nesEs . chainNes $ chainSt
-       in balance u <+> d <+> f
+       in Val.coin (balance u) <+> d <+> f
 
```

Comment:
same as above - should this add up all the assets, not just check that coins add up correctly?